src/xpk/commands/batch.py

```diff
@@ -0,0 +1,59 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from argparse import Namespace
+from ..utils import xpk_exit, xpk_print
+from .cluster import set_cluster_command
+from ..core.core import add_zone_and_project
+from ..core.job_template import create_job_template_instance
+from ..core.app_profile import create_app_profile_instance
+from ..core.app_profile import APP_PROFILE_TEMPLATE_DEFAULT_NAME
+from ..core.commands import (
+    run_command_for_value,
+)
+
+
+def batch(args: Namespace) -> None:
+  """Run batch task.
+     This function runs passed script in non-blocking manner.
+  Args:
+    args: user provided arguments for running the command.
+  Returns:
+    None
+  """
+  add_zone_and_project(args)
+  set_cluster_command_code = set_cluster_command(args)
+  if set_cluster_command_code != 0:
+    xpk_exit(set_cluster_command_code)
+
+  create_job_template_instance(args)
+  create_app_profile_instance(args)
+  submit_job(args)
+
+
+def submit_job(args: Namespace) -> None:
+  cmd = (
+      'kubectl-kjob create slurm --profile'
+      f' {APP_PROFILE_TEMPLATE_DEFAULT_NAME} --'
+      f' {args.script}'
+  )
+  if len(args.script_args) != 0:
+    cmd += f'{args.script_args}'
+  return_code, _ = run_command_for_value(cmd, 'submit job', args)
+
+  if return_code != 0:
+    xpk_print(f'Running batch job returned ERROR {return_code}')
+    xpk_exit(return_code)
```

Comment:
I'm just wondering if we should use xpk_exit() in the indirect functions (functions that are not explicitly called by parser). It doesn't seem to be the case in the rest of the code - such functions usually print out some info and return an error code which is then processed by the upper most function.

I don't necessarily agree with the way it is done currently. The way error codes are processed is usually just a single "if" followed by exit function. Still it gives us a guarantee that the code won't randomly exit when calling some function. I would like to keep this guarantee for now, so please return the error code or 0 and check for it in the batch() function.

Reply:
I think it makes neccessary flow of error codes. If cmd execution exists with error we should exit immediately.

Reply:
Alright, but let's keep in mind that we should also in the future change the existing code to this convention.